### PR TITLE
fix: modal animations, new fade animation

### PR DESCRIPTION
**Pull request by just-toby**
our feedback is that these modals should behave [exactly like the other modals in the app](https://www.notion.so/uniswaplabs/The-modal-animation-when-you-close-it-feels-jumpy-sudden-6eea1bfb09964247ac958a74f2b93b9b?pvs=4)

this means removing the y-translate and scaling, and just keeping an opacity transition. also, this fixes the animation to apply to the background scrim too, so it feels less jumpy

### CODE REVIEW

File: src/components/Dialog.tsx [hunk 1]
```diff
@@ -243,17 +249,12 @@ const getAnimation = (animationType?: DialogAnimationType) => {
...
-`
-
-const FullScreenWrapper = styled.div<{ enabled?: boolean }>`
-  ${({ enabled }) =>
+const FullScreenWrapper = styled.div<{ enabled?: boolean; animationType?: DialogAnimationType }>`
+  ${({ enabled, animationType }) =>
     enabled &&
     css`
       align-items: center;
+      ${animationType === DialogAnimationType.FADE ? fadeAnimationCss : ''}
```
Comment (just-toby) [inline src/components/Dialog.tsx:257]: `FullScreenWrapper` is the backdrop for full screen modals - this css is only applied if we enable the backdrop via the prop

File: src/components/Dialog.tsx [hunk 2]
```diff
@@ -186,13 +186,19 @@ const slideOutRight = keyframes`
 
 const fadeIn = keyframes`
```
Comment (zzmp) [inline src/components/Dialog.tsx:187]: These are now generic enough that I think `fadeIn`/`fadeOut`/`fadeAnimationCss` can all be moved to `theme/animations.ts`.

File: src/components/Dialog.tsx (hunk 1 above)
Comment (zzmp) [inline src/components/Dialog.tsx:257]: Are there other `DialogAnimationType`s, or should this be a `boolean`?

File: src/components/Dialog.tsx [hunk 3]
```diff
@@ -339,7 +348,12 @@ export default function Dialog({ color, children, onClose, forceContain }: Dialo
       <ThemeProvider>
         <PopoverBoundaryProvider value={popoverRef.current} updateTrigger={updatePopover}>
           <div ref={popoverRef}>
-            <FullScreenWrapper enabled={pageCentered} onClick={closeOnBackgroundClick}>
+            <FullScreenWrapper
+              enabled={pageCentered}
+              onClick={closeOnBackgroundClick}
+              ref={fullScreenWrapperRef}
+              animationType={context.options?.animationType}
```
Comment (zzmp) [inline src/components/Dialog.tsx:355]: Should this be a boolean - `fade={context.options?.animationType === DialogAnimationType.FADE}`?

File: src/components/Dialog.tsx [hunk 4]
```diff
@@ -327,7 +336,7 @@ export default function Dialog({ color, children, onClose, forceContain }: Dialo
           return (mountPoint?.childElementCount ?? 0) > 1 ? SlideAnimationType.PAGING : SlideAnimationType.CLOSING
       }
     },
-    modal,
+    [fullScreenWrapperRef, modal],
```
Comment (zzmp) [inline src/components/Dialog.tsx:339]: Would it be safer to use nesting CSS instead of passing an array here?

**Review by zzmp — APPROVED** [review]
lgtm - there are minor nits that could simplify the code, but nothing blocking, and the functionality looks good.

File: src/components/Dialog.tsx (hunk 4 above)
Comment (just-toby) [reply src/components/Dialog.tsx:304]: i think this is worth looking into later. i actually tried that approach at first but couldn't get it working. can't spend more time on this atm but i'll circle back.

**Pull request merged**